src/Adapter/MSTest.TestAdapter/Discovery/AssemblyEnumerator.cs

```diff
@@ -354,60 +366,38 @@ private static bool TryProcessTestDataSourceTests(UnitTestElement test, TestMeth
         }
     }
 
-    private static bool ProcessTestDataSourceTests(UnitTestElement test, MethodInfo methodInfo, UTF.ITestDataSource[] testDataSources, List<UnitTestElement> tests)
+    private static bool ProcessTestDataSourceTests(UnitTestElement test, MethodInfo methodInfo, FrameworkITestDataSource[] testDataSources,
+        List<UnitTestElement> tests)
     {
         foreach (var dataSource in testDataSources)
         {
             var data = dataSource.GetData(methodInfo);
-            var discoveredTests = new Dictionary<string, UnitTestElement>();
-            var discoveredIndex = new Dictionary<string, int>();
-            var serializationFailed = false;
             var index = 0;
 
             foreach (var d in data)
             {
                 var discoveredTest = test.Clone();
                 discoveredTest.DisplayName = dataSource.GetDisplayName(methodInfo, d) ?? discoveredTest.DisplayName;
 
-                // if we have a duplicate test name don't expand the test, bail out.
```

Comment:
Maybe I should keep this logic for the "DisplayName" strategy.

Reply:
If we remove this, it will cause error in some scenarios. If we're removing bail-out, we should hard fail in that scenario.

Reply:
Reverting this part.